editor/header/tools/publish-button.js

```diff
@@ -61,15 +70,13 @@ function PublishButton( {
 
 export default connect(
 	( state ) => ( {
-		post: state.currentPost,
-		edits: state.editor.edits,
-		blocks: state.editor.blockOrder.map( ( uid ) => (
-			state.editor.blocksByUid[ uid ]
-		) ),
-		isRequesting: state.saving.requesting,
-		isSuccessful: state.saving.successful,
-		isError: !! state.saving.error,
-		requestIsNewPost: state.saving.isNew,
+		post: getCurrentPost( state ),
+		edits: getPostEdits( state ),
+		blocks: getBlocks( state ),
+		isRequesting: isSavingPost( state ),
+		isSuccessful: isPostSaveRequestSuccessful( state ),
+		isError: !! isPostSaveRequestFailed( state ),
```

Comment:
We can assume `isPostSaveRequestFailed` will return a boolean and drop the coercion